scipy/stats/_distribution_infrastructure.py

```diff
@@ -3684,7 +3785,7 @@ def logintegrand(x, **params):
             # so logpmf is always negative, and so log(logpmf) = log(-logpmf) + pi*j.
             # The two imaginary components "cancel" each other out (which we would
             # expect because each term of the entropy summand is positive).
-            return logpmf + np.log(-logpmf)
+            return np.where(np.isfinite(logpmf), logpmf + np.log(-logpmf), -np.inf)
```

Comment:
I needed to do this to get the tests to pass for degenerate cases where binomial is a point distribution. This is the usual convention that when calculating entropy, `p * log(p)` is considered to be `0` when `p = 0`.